### ISSUE TITLE
Docker image `v0.3.0` might need to be rebuilt

### ISSUE THREAD

**Issue by LorenzoFernando** [DirectXMan12/k8s-prometheus-adapter#126]
# tldr

The official `v0.3.0` image might need to be recreated. 

I recompiled the image and pushed to my public repository [here](https://hub.docker.com/r/lorenzofernando/k8s-prometheus-adapter-amd64/tags/) after noticing these log entries indicating issues that were addressed in the `v0.3.0` release:

```
W1110 20:16:23.588165       1 authentication.go:245] Unable to get configmap/extension-apiserver-authentication in kube-system.  Usually fixed by 'kubectl create rolebinding -n kube-system ROLE_NAME --role=extension-apiserver-authentication-reader --serviceaccount=YOUR_NS:YOUR_SA'
F1110 20:16:23.588358       1 adapter.go:206] unable to install resource metrics API: open /tmp/client-ca-file507372049: no such file or directory
```

## The Fix
Recreating the docker image with a minor change to the Makefile to address a `sed` API incompatibility with `macos` (see the diff below), tagging it `recompile`, and updating the `prometheus-adapter` Deployment, I was able to resolve the issue. Let me know if I can provide any additional information. I spent a lot of time investigating RBAC issues in the `prometheus-operator` repository and I don't think the cause lies there.

To be clear, this is concerning the image at `quay.io/coreos/k8s-prometheus-adapter-amd64:v0.3.0`. If that is outside your control then I will close this issue. 

Thanks!

## Recreate the issue
```shell
minikube start --kubernetes-version v1.12.2 --cpus 6 --memory 8192
git clone git@github.com:coreos/prometheus-operator.git
cd ./prometheus-operator/contrib/kube-prometheus
kubectl create -f manifests/ || true
kubectl create -f manifests/ || true # Run twice to address a race condition with resource creation
kubectl get pods -n monitoring | grep prometheus-adapter # Here you should see the crashloop pod condition
```

## Change made to this repository
```diff
diff --git a/Makefile b/Makefile
index 935a0bb..8e6ae45 100644
--- a/Makefile
+++ b/Makefile
@@ -37,7 +37,7 @@ $(OUT_DIR)/%/adapter: $(src_deps)

 docker-build:
        cp deploy/Dockerfile $(TEMP_DIR)
-       cd $(TEMP_DIR) && sed -i "s|BASEIMAGE|$(BASEIMAGE)|g" Dockerfile
+       cd $(TEMP_DIR) && sed -i- "s|BASEIMAGE|$(BASEIMAGE)|g" Dockerfile && rm Dockerfile-

        docker run -it -v $(TEMP_DIR):/build -v $(shell pwd):/go/src/github.com/directxman12/k8s-prometheus-adapter -e GOARCH=$(ARCH) $(GOIMAGE) /bin/bash -c "\
                CGO_ENABLED=0 go build -tags netgo -o /build/adapter github.com/directxman12/k8s-prometheus-adapter/cmd/adapter"
@@ -48,7 +48,7 @@ docker-build:
 build-local-image: $(OUT_DIR)/$(ARCH)/adapter
        cp deploy/Dockerfile $(TEMP_DIR)
        cp  $(OUT_DIR)/$(ARCH)/adapter $(TEMP_DIR)
-       cd $(TEMP_DIR) && sed -i "s|BASEIMAGE|scratch|g" Dockerfile
+       cd $(TEMP_DIR) && sed -i- "s|BASEIMAGE|scratch|g" Dockerfile && rm Dockerfile-
        docker build -t $(REGISTRY)/$(IMAGE)-$(ARCH):$(VERSION) $(TEMP_DIR)
        rm -rf $(TEMP_DIR)

(END)```

**Comment by LorenzoFernando** [comment]
1. I managed to find the repo owner's (i.e. `DirectXMan12`) docker repository, and after some brief testing found that their image does *NOT* have this issue. The issue seems to be solely with the quay.io/coreos repository. 

2. The changes to the `Makefile` I included in my first message might still be a good idea. I referenced this issue when making my changes: https://github.com/kubernetes/dns/pull/87

**Comment by webwurst** [comment]
@LorenzoFernando thanks a lot for pointing this out! I thought the error was caused by misconfigured `RBAC` rules, but couldn't find any problems there. I can confirm that the image `directxman12/k8s-prometheus-adapter-amd64:v0.3.0` does not show this error.

**Comment by s-urbaniak** [comment]
I can reproduce this here. The problem is, that there is no `/tmp` tmpfs mount available in the running container. This probably used to work for docker, when used as the container runtime, but apparently other CRI runtimes (i.e. containerd, etc.) are less magic, i.e. they don't mount `/tmp` tmpfs by default.

@LorenzoFernando can you add a `emptyDir` mount to `/tmp` in your adapter deployment as described in https://kubernetes.io/docs/concepts/storage/volumes/#emptydir and recheck?

**Comment by metalmatze** [comment]
After merging the changes into kube-prometheus on Friday I can confirm, that the image didn't work for Docker either. I was running a [kind](https://github.com/kubernetes-sigs/kind) cluster locally for testing.
Thanks for looking into this!

**Comment by metalmatze** [comment]
Apparently there was a mistake by building the CoreOS image locally. We have now copied the upstream directxman12 docker hub image and it works.
Please close this issue, if it works for you now.
Sorry for the inconvenience!
